### PR TITLE
feat(web): Add relative url mapping so CMS links can point to prod and code makes them relative

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Add relative url mapping so CMS links can point to prod and code makes them relative

## What

For E2E tests to not leave the environment they are testing once they navigate using a link, this was deemed the best approach. Instead of relying on CMS users to create relative links we can instead handle it so if they create an absolute link to prod then we'll automatically make that link relative

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/cms/src/lib/models/utils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const prefix = 'https://island.is'
+
+/** Make sure that links are relative in case someone links to production directly */
+export const getRelativeUrl = (url: string) => {
+  const trimmedUrl = url.trim()
+
+  if (trimmedUrl.startsWith(prefix)) {
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply libs/cms/src/lib/models/utils.ts]: I've pushed new code that uses the `new URL()` method instead.

File: libs/cms/src/lib/models/utils.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/** Make sure that links are relative in case someone links to production directly */
+export const getRelativeUrl = (url: string) =>
+  new URL(url.trim(), 'https://island.is').pathname
```
Comment (AndesKrrrrrrrrrrr) [inline libs/cms/src/lib/models/utils.ts:3]: Could maybe remove all hosts? This is indeed the most common, even the only host in our urls. But what if someone accidentally adds `http://localhost` or `https://my-feature.dev01.devland.is`?

Comment (RunarVestmann) [reply libs/cms/src/lib/models/utils.ts:3]: Hmm, ok so if the host is similar to the check made here: 
image

Then we should remove it, to cover the cases where a link is perhaps pointing to dev? What if a link is supposed to be pointing to a feature deployed link, would we still want to overwrite it and make it relative?

Comment (RunarVestmann) [reply libs/cms/src/lib/models/utils.ts]: I noticed an error in my logic I was basically making external links be relative (by joinking their pathname) so I fixed that

**Review by Ingimarsson — APPROVED** [review]

**Pull request merged**